pkg/scraper/aws/client.go

```diff
@@ -18,11 +19,12 @@ import (
 	"k8s.io/client-go/rest"
 )
 
-func NewAWSClient(ctx context.Context, profile, region string) (interfaces.AWSClient, error) {
+func NewAWSClient(ctx context.Context, profile, region, roleARN string) (interfaces.AWSClient, error) {
```

Comment:
maybe profile and roleARN should be options you pass in like:

~~~go
func NewAWSClient(ctx context.Context, opts ...AWSClientOpt) (interfaces.AWSClient, error) {
~~~

otherwise you end up passing a bunch of empty strings around your constructors

Reply:
and its either one or the other. If you don't want to use that, maybe have separate constructors. One for profile and one of roleARN: `NewAssumeRoleAWSClient` and `NewProfileAWSClient`

Reply:
Makes sense